iiod/iiod.c

```diff
@@ -414,6 +415,12 @@ int main(int argc, char **argv)
 
 			uart_params = optarg;
 			break;
+		case 'p':
+			val = strtoul(optarg, &end, 0);
+			if (optarg == end || errno == ERANGE || val > 0xFFFF)
```

Comment:
A few comments here:
- You can probably parse the value in base 10, I don't think anybody sane uses hexadecimal for network ports.
- I would believe that if you get a bogus value, you should return an error instead of silently falling back to the default port.
- Your errno check is invalid, since errno may already be ERANGE before the call to strtoul. To verify that the conversion failed it should be enough to verify that `optarg == end || *end == '\0'` (plus the range check).